chat/views.py

```diff
@@ -1,18 +1,124 @@
+            "is_group": true,
+            "participants": ["user1", "user2", "user3"]
+        }
+
+    Response example:
+        {
+            "name": "friends_group",
+            "is_group": true,
+            "participants": ["user1", "user2", "user3"]
+        }
+    """
+    permission_classes = [IsAuthenticated]
+    serializer_class = RoomSerializer
+
+
+class SendMessageView(APIView):
+    """
+    Send a new message within a conversation and broadcast it via WebSocket.
+
+    Endpoint:
+        POST /api/messages/
+
+    Request body example:
+        {
+            "room": "friends_group",
+            "sender_id": "user1",
+            "text": "Hello everyone!"
+        }
+
+    Response example:
+        {
+            "room": "friends_group",
+            "sender_id": "user1",
+            "receiver_id": null,
+            "text": "Hello everyone!",
+            "timestamp": "2025-08-25T20:00:00Z",
+            "is_read": false
+        }
+    """
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, *args, **kwargs):
+        serializer = MessageSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        message = serializer.save()
+
+        channel_layer = get_channel_layer()
+        room_name = message.room.name
+        data = MessageSerializer(message).data
+
+        async_to_sync(channel_layer.group_send)(
+            f"chat_{room_name}",
+            {"type": "chat_message", "message": data}
+        )
+
+        return Response(data, status=status.HTTP_201_CREATED)
+
 
+class ConversationMessagesView(generics.ListAPIView):
+    """
+    Retrieve the list of messages in a conversation.
 
-def index(request):
-    return render(request, "chat/index.html")
+    Endpoint:
+        GET /api/conversations/{room_name}/messages/
 
+    Path parameter:
+        room_name (str): Name of the conversation/room.
 
-def room(request, room_name):
-    return render(request, "chat/room.html", {"room_name": room_name})
+    Response example:
+        [
+            {
+                "room": "friends_group",
+                "sender_id": "user1",
+                "receiver_id": null,
+                "text": "Hello everyone!",
+                "timestamp": "2025-08-25T20:00:00Z",
+                "is_read": false
+            },
+            {
+                "room": "friends_group",
+                "sender_id": "user2",
+                "receiver_id": "user1",
+                "text": "Hi!",
+                "timestamp": "2025-08-25T20:01:00Z",
+                "is_read": false
+            }
+        ]
+    """
+    permission_classes = [IsAuthenticated]
+    serializer_class = MessageSerializer
+    pagination_class = LimitOffsetPagination
 
+    def get_queryset(self):
+        room_name = self.kwargs["room_name"]
+        try:
+            room = Room.objects.get(name=room_name)
+        except Room.DoesNotExist:
+            return Message.objects.none()
```

Comment:
Returning Message.objects.none() for missing room does not distinguish error from empty messages. Return a 404 response when room does not exist.
```
if not Room.objects.filter(name=room_name).exists():
    raise Http404("Room not found")
```